### PR TITLE
bugfix: avoid nilerr cases happen in ci-lint (#610)

### LINKED ISSUE #610: [nilerr] there are some nilerr cases in sealer which needs fixed
#### What happened:
when we apply golangci-lint with rule nilerr https://github.com/gostaticanalysis/nilerr, we found that some places needed updated.

```
image/store/distribution.go:44:3: error is not nil (line 42) but it returns nil (nilerr)
		return res, nil
		^
utils/ssh/scp.go:340:3: error is not nil (line 339) but it returns nil (nilerr)
		return false, nil
```

#### What you expected to happen:
we wish no nilerr case happens.

#### How to reproduce it (as minimally and precisely as possible):
none

#### Anything else we need to know?:
none

#### Environment:
- sealer version (use `sealer version`): main branch
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by neverCase**
### Does this pull request fix one issue?

Fixes #610

### CODE REVIEW

File: image/store/distribution.go [hunk 1]
```diff
@@ -41,7 +41,8 @@ func (fs *filesystem) LoadDistributionMetadata(layerID LayerID) (map[string]dige
 	)
 	distributionMetadataFile, err := os.Open(filepath.Join(layerDBPath, "distribution_layer_digest"))
 	if err != nil {
-		return res, nil
+		//lint:ignore nilerr https://github.com/alibaba/sealer/issues/610
```
Comment (fanux) [inline image/store/distribution.go:44]: I think we need return error here,not ignore it~

File: utils/ssh/scp.go [hunk 2]
```diff
@@ -337,7 +337,8 @@ func (s *SSH) RemoteDirExist(host, remoteDirpath string) (bool, error) {
 	}
 	defer sftpClient.Close()
 	if _, err := sftpClient.ReadDir(remoteDirpath); err != nil {
-		return false, nil
+		//lint:ignore nilerr https://github.com/alibaba/sealer/issues/610
```
Comment (fanux) [inline utils/ssh/scp.go:340]: return error if not nil

File: image/store/distribution.go (hunk 1 above)
Comment (fanux) [reply image/store/distribution.go:44]: @justadogistaken double check this~

Comment (justadogistaken) [reply image/store/distribution.go:44]: There would be some problems, distribution_layer_digest does not exist for all layers, which will be generated by "pull"/"push", but "build". If we return err here, layer can't be loaded.

Comment (justadogistaken) [reply image/store/distribution.go:44]: Maybe we should return error here, it's just a low-level func. But we need to ignore the error at the caller side

Comment (fanux) [reply image/store/distribution.go:44]: `ignore the error at the caller side`, agree~

Comment (neverCase) [reply image/store/distribution.go:44]: so it's my turn to take next pr, right.
which steps should i choose:
- create a new branch from main and modify the codes 
- modify the codes based on the last pr branch
@fanux

Comment (fanux) [reply image/store/distribution.go:44]: > so it's my turn to take next pr, right.
> which steps should i choose:
> 
> * create a new branch from main and modify the codes
> * modify the codes based on the last pr branch
>   @fanux

Just modify this branch and push~

**Comment by allencloud** [comment]
We merge this first? @fanux

**Comment by fanux** [comment]
@allencloud This PR still has a litter bit problem

File: image/store/distribution.go [hunk 3]
```diff
@@ -41,7 +41,7 @@ func (fs *filesystem) LoadDistributionMetadata(layerID LayerID) (map[string]dige
 	)
 	distributionMetadataFile, err := os.Open(filepath.Join(layerDBPath, "distribution_layer_digest"))
 	if err != nil {
-		return res, nil
+		return res, err
```
Comment (fanux) [inline image/store/distribution.go:44]: ignore this error, build will failed if return error here.

File: image/store/distribution.go (hunk 1 above)
Comment (neverCase) [reply image/store/distribution.go]: 👌 , I got it wrong

**Review by fanux — APPROVED** [review]

**Pull request merged**